tools/proofers/for_mentors.php

```diff
@@ -150,10 +150,16 @@
 
 // XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 
-function output_project_label($nameofwork, $authorsname)
+function output_project_label($nameofwork, $authorsname, $date = null)
 {
-    // TRANSLATORS: format is <title> by <author>.
-    echo sprintf("%1\$s by %2\$s", $nameofwork, $authorsname);
+    if (!is_null($date)) {
+        $date = sprintf("(" . ("%s"), strftime("%Y-%m-%d %R %Z", $date) . ")");
```

Comment:
Could you write this as

```php
    $date = strftime("%Y-%m-%d %R %Z", $date);
    // TRANSLATORS: format is <title> by <author> (<date>).
    echo sprintf(_("%1\$s by %2\$s (%3\$s)"), $nameofwork, $authorsname, $date);
```

Reply:
Hm ... yes, I probably should have done it that way. Thinking back, I think I was initially trying to figure out how to build the date so that I wouldn't need the conditional if the date is `null`, but decided I was better off just including it. And then forgot. Thanks!

Reply:
Done.

Reply:
In this case I would just append the date to the same string and have one fewer strings the translators need to translate:
```php
    // TRANSLATORS: format is <title> by <author>.
    echo sprintf("%1\$s by %2\$s", $nameofwork, $authorsname);
    if (!is_null($date)) {
        $date = strftime("%Y-%m-%d %R %Z", $date);
        echo " ($date)";
    }
```

Reply:
Thanks, that simplifies it quite a bit! One quick question, though -- doesn't the `"%1\$s by %2\$s"` need to be wrapped in `_()`, or am I missing something?

Reply:
It sure should!